### PR TITLE
convert publishing and unpublishing of collection products to graphql

### PR DESCRIPTION
**Pull request by tekhaus**

### CODE REVIEW

**Comment by tekhaus** [comment]
[Modernize the publish-/unpublish-a-collection tasks](https://trello.com/c/YCSB2obP/9-modernize-the-publish-unpublish-a-collection-tasks)

File: docs/publish-a-certain-collection-daily/script.liquid [hunk 1]
```diff
@@ -1,37 +1,73 @@
...
-        "update",
-        [
-          "product",
-          12345
-        ],
-        {
-          "published": true
+{% assign collection = shop.collections[options.collection_id__required] %}
+
+{% if collection == blank and event.preview == false %}
```
Comment (isaacbowen) [inline docs/publish-a-certain-collection-daily/script.liquid:3]: Nuance: `event.preview` is never false ([docs](https://docs.usemechanic.com/article/373-the-event-object)). It's either true, or the `event` object just doesn't have a `preview` property at all. I think the best way to express the intent here is `event.preview != true`.

File: docs/unpublish-a-certain-collection-daily/script.liquid [hunk 2]
```diff
@@ -1,37 +1,73 @@
...
-        "update",
-        [
-          "product",
-          12345
-        ],
-        {
-          "published": false
+{% assign collection = shop.collections[options.collection_id__required] %}
+
+{% if collection == blank and event.preview == false %}
```
Comment (isaacbowen) [inline docs/unpublish-a-certain-collection-daily/script.liquid:3]: ditto

**Review by isaacbowen — CHANGES_REQUESTED** [review]
Good! Thank you! :D One adjustment, then we're clear. 👍

**Review by isaacbowen — APPROVED** [review]

**Pull request merged**